Aquarius/Src/Aquarius/Core/Window.cpp

```diff
@@ -0,0 +1,109 @@
+#include "Window.h"
+#include "Log.h"
+#include <assert.h>
+
+namespace Aquarius {
+
+    Window::Window(uint32_t width, uint32_t height, std::string &&name, bool vsync)
+        : m_Width(width), m_Height(height), m_Name(std::move(name)), m_VsyncEnabled(vsync), m_Window(nullptr), m_Context(nullptr)
+    {}
+
+    Window::WindowPtr Window::Create(uint32_t width, uint32_t height, std::string &&name, bool vsync)
+    {
+        return std::make_unique<Window>(width, height, std::move(name));
+    }
+
+    void Window::Initialize()
+    {
+        // Initialize GLFW
+        int initSuccess = glfwInit();
+        if (!initSuccess)
+        {
+            AQ_CORE_FATAL("Failed to initialize GLFW");
+        }
+        else
+        {
+            AQ_CORE_INFO("GLFW successfully initialized");
+        }
+
+        glfwWindowHint(GLFW_CONTEXT_VERSION_MAJOR, 4);
+        glfwWindowHint(GLFW_CONTEXT_VERSION_MINOR, 1);
+        glfwWindowHint(GLFW_OPENGL_PROFILE, GLFW_OPENGL_CORE_PROFILE);
+
+        #ifdef PLATFORM_MAC
+                glfwWindowHint(GLFW_OPENGL_FORWARD_COMPAT, GL_TRUE);
+        #endif // PLATFORM_MAC
+
+        m_Window = glfwCreateWindow(m_Width, m_Height, m_Name.c_str(), NULL, NULL);
+        if (m_Window == nullptr)
+        {
+            AQ_CORE_FATAL("Failed to create a glfw window");
+            glfwTerminate();
+            assert(false);
```

Comment:
Might want to double check, but I believe AQ_CORE_FATAL will cause program termination, in that case, may wanna remove the assert and move the call to `glfwTerminate()` to above the log that was all the opengl stuff exits cleanly before terminate

Reply:
Didn't realize fatal would terminate until after this, good point